Fix bug of negative target temperature in visu screen Anycubic I3 MEGA serie

stef-ladefense opened the pull request:
the target temperature is positive, the convertion is unsigned.

### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description

<!--

the display is incorrect with the requested temperature values. it displays in the form -xx when it exceeds 128 ° C because the conversion used is signed.

-->

### Benefits

<!-- fix with unsigned conversion
[stb.zip](https://github.com/MarlinFirmware/Marlin/files/5312831/stb.zip)
 -->

### Configurations

<!-- Attach any Configuration.h, Configuration_adv.h, or platformio.ini files needed to compile/test your Pull Request. -->

### Related Issues

<!-- Whether this fixes a bug or fulfills a feature request, please list any related Issues here. -->

stef-ladefense committed:
Fix bug of negative target temperature

the target temperature is positive, the convertion is unsigned.

thisiskeithb commented:
Does this fix https://github.com/MarlinFirmware/Marlin/issues/19564 ?

sjasonsmith commented:
Does this have to remain 8 bit? That will cause it to roll from 255 back to 0. I assume the display should be able to handle any 3-digit number, so a 16 bit integer should be used?

stef-ladefense commented:
> Does this have to remain 8 bit? That will cause it to roll from 255 back to 0. I assume the display should be able to handle any 3-digit number, so a 16 bit integer should be used?

I agree with you, here given the code of the function (which was in marlin already, I did nothing but change to a usigned) the maximum value is 255, so yes it would be necessary to use a 16-bit function as input, always on 3 digits to display greater than 255. can use ui16tostr3rj for this, i test it !

there in this case on the V5 head, it is strongly advised not to exceed 250 ° C otherwise the Teflon of the brodwen melts.

stef-ladefense committed:
Update anycubic_i3mega_lcd.cpp

stef-ladefense commented:
it works very well by replacing "ui8tostr3rj" by "ui16tostr3rj"
it will support heads that support over 255 ° C.

on the other hand I do not know how to change the PR without making a new one ?

thisiskeithb commented:
> on the other hand I do not know how to change the PR without making a new one ?

You just push changes to your PR branch, but you’ve already done that, so you’re good to go.

stef-ladefense commented:
> > on the other hand I do not know how to change the PR without making a new one ?
> 
> You just push changes to your PR branch, but you’ve already done that, so you’re good to go.

very well then, thank you
I don't know how to update my Marlin folk according to the PR applied to the repository.
I haven't figured that out yet, for now I'm destroying my folk, begging for folk Marlin, to keep up to date.
it's not easy with the git manadger

sjasonsmith commented:
Ok, one more thing. That should probably be a signed 16-bit integer, since temperatures can read negative if the thermistor is not connected.

stef-ladefense commented:
> Ok, one more thing. That should probably be a signed 16-bit integer, since temperatures can read negative if the thermistor is not connected.

love you ! you are nices ideas !

I just tested with "i16tostr3rj" it also works,
I disconnected TE0 and the output goes to 0 ° C
no negative in my case
a pro Marlin will certainly be able to tell what is happening and if the temperature value of the probe can be negative

stef-ladefense committed:
Update anycubic_i3mega_lcd.cpp

stef-ladefense committed:
FIX directory not work on SD card

with longFilename, the directory appears just with a '\' without anything behind

thinkyhead commented:
It is still a mystery as to why the long names are coming up empty in the file listing.

rhapsodyv commented:
> It is still a mystery as to why the long names are coming up empty in the file listing.

They are really long names? I found that if the name fit in 8.3, windows could save the file without long name flag.

I found it when working in the lvgl asset loader. When long name is empty, I fallback to the dos name and it works fine.

Maybe we should do it in the core.
